Shared_KKEC/Clothes/KoiClothesOverlayController.Hooks.cs

```diff
@@ -216,6 +217,91 @@ public static Texture GetMask(KoiClothesOverlayController controller, MaskKind k
 
             #endregion
 
+            #region Colormasks
+            [HarmonyPostfix]
+            [HarmonyWrapSafe]
+            [HarmonyPatch(typeof(ChaControl), nameof(ChaControl.InitBaseCustomTextureClothes))]
+            public static void InitBaseCustomTextureClothesPostHook(ChaControl __instance, bool main, int parts)
+            {
+                var updated = false;
+                var clothesId = GetClothesIdFromKind(main, parts);
+                clothesId = main ? MakeColormaskId(clothesId) : MakeColormaskId(clothesId);
+
+                var registration = CharacterApi.GetRegisteredBehaviour(typeof(KoiClothesOverlayController));
+                if (registration == null) throw new ArgumentNullException(nameof(registration));
+                foreach (var controller in registration.Instances.Cast<KoiClothesOverlayController>()) {
+                    for (int i = 0; i < 3; i++)
+                    {
+                        var tex = controller.GetOverlayTex(clothesId, false)?.Texture;
+                        if (tex != null)
+                        {
+                            if (main && parts < __instance.ctCreateClothes.GetLength(0) && i < __instance.ctCreateClothes.GetLength(1) && __instance.ctCreateClothes[parts, i] != null)
+                            {
+                                updated = true;
+                                __instance.ctCreateClothes[parts, i].SetTexture(ChaShader._ColorMask, tex);
+                            }
+                            else if (parts < __instance.ctCreateClothesSub.GetLength(0) && i < __instance.ctCreateClothesSub.GetLength(1) && __instance.ctCreateClothesSub[parts, i] != null)
+                            {
+                                updated = true;
+                                __instance.ctCreateClothesSub[parts, i].SetTexture(ChaShader._ColorMask, tex);
+                            }
+                        }
+                    }
```

Comment:
I don't understand this part. Why are you iterating on every character in a single character's init method? What if there's only one character, or if there's 100? Can't you do `__instance.GetComponent<KoiClothesOverlayController>()` and do your work with it?